apps/files/utils.py

```diff
@@ -542,6 +542,10 @@ def parse_xpi(xpi, addon=None, check=True):
 def check_xpi_info(xpi_info, addon=None):
     from addons.models import Addon, BlacklistedGuid
     guid = xpi_info['guid']
+    if amo.get_user():
+        user_id = amo.get_user().id
+    else:
+        user_id = 0
```

Comment:
We still need to check deleted addons for conflicting GUIDs, even if there is no user.  Its basically `Addon.unfiltered.filter(guid=guid).exists())` in that case.  I had it written like that at first but switched it to fake user id to reduce the complexity a little but if its more explicit I can change back.
I'm pretty sure amo.get_user() only returns None in tests (some were failing) so I special cased it.